### DISCUSSION TITLE
NuMojo V0.2

### DISCUSSION THREAD

**Discussion by MadAlex1997** [Mojo-Numerics-and-Algorithms-group/NuMojo#68] [General]
## Effort 1
Getting and setting methods.
Features:
* `__getitem__` and  `__setitem__` for boolean NDArrays
    *  Get item should return a flat array of values for which the boolean values are true.
    * Set item should set the items where the mask array is true to the scalar value on the left-hand side.
* `__getitem__` (or potentially non-dunder)  method for a List of indexes returns a flattened array
    * How would it return a `ndim`>1 array? 

## Effort 2
Cleaning up code.
Features:
* Modify the codebase to follow the style guide.

## Effort 3
Standardize math for `NDArray`.
Features:
* Modify dunder methods to use `math` module functions.
* Add `math` `Backend` parameter as a keyword-only parameter for `NDArray`
* Possibly adding more math features and dunder methods.

## Effort 4 (Long term)
Start thinking about `StaticNDArray`.
## Note:
@MadAlex1997 will take primary on Effort 3.

**Comment by forfudan** [comment]
I will dive into Effort 1(b) and Effort 2.

**Comment by shivasankarka** [comment]
I'll work on tasks in Effort 1 and Effort 2. 

I have already worked on StaticNDArray and ported most of the basic things to compile time, so I could take that branch and modify it to match our current implementation, So I can work on Effort (4) slowly on the side. 

Sounds like a plan!
